src/components/c-tabs.vue

```diff
@@ -0,0 +1,202 @@
+    <ul :class="b('tab-list')"
+        :aria-label="ariaLabel"
+        role="tablist"
+    >
+      <li v-for="tab in tabs"
+          :key="tab.id"
+          :class="b('tab-item')"
+      >
+        <a :id="`c-tabs-${uuid}--tab-${tab.id}`"
+           :class="b('tab', { [tab.id]: true, active: tab.id === activeTab.id })"
+           :aria-selected="tab.id === activeTab.id"
+           :aria-controls="`c-tabs-${uuid}--panel-${tab.id}`"
+           :href="tab.link?.href || `#${tab.id}`"
+           :target="tab.link?.target || LINK_TARGET.SELF"
+           :rel="tab.link?.target === LINK_TARGET.BLANK && 'noopener noreferrer'"
+           role="tab"
+           @click="onTabClick(tab)"
+        >
+          <!-- @slot ${tab.id}__tab - Allows to overwrite the tab content. -->
+          <slot :name="`${tab.id}__tab`">
+            {{ tab.title }}
+          </slot>
+        </a>
+      </li>
+    </ul>
+    <!-- @slot default - Allows to use the component without panels. -->
+    <slot>
+      <div v-for="tab in tabs"
+           :id="`c-tabs-${uuid}--panel-${tab.id}`"
+           :key="tab.id"
+           :class="b('panel', { [tab.id]: true, visible: tab.id === activeTab.id })"
+           :aria-labelledby="`c-tabs-${uuid}--tab-${tab.id}`"
+           :tabindex="tab.id === activeTab.id ? 0 : -1"
+           role="tabpanel"
+      >
+        <!-- @slot ${tab.id} - Renders the panel content for the current tab. -->
+        <slot :name="tab.id"></slot>
+      </div>
+    </slot>
+  </div>
+</template>
+
+<script lang="ts">
+  import { defineComponent, PropType } from 'vue';
+  import useUuid, { IUuid } from '@/compositions/uuid';
+  import { LINK_TARGET } from '@/setup/globals';
+  import { ILink } from '@/types/link';
+
+  interface ISetup extends IUuid {}
+
+  export interface ITab {
+
+    /**
+     * The text title for the current tab.
+     */
+    title: string,
+
+    /**
+     * A unique id for the current tab.
+     */
+    id: string,
+
+    /**
+     * Determines if the given tab should be initially active.
+     */
+    active?: boolean,
+
+    /**
+     * Allows to track additional information for a tab. e.g. a link.
+     */
+    link?: ILink,
+  }
+
+  interface IData {
+    LINK_TARGET: typeof LINK_TARGET,
+
+    /**
+     * Holds the currently active tab defintion.
+     */
+    activeTab: ITab|null,
+  }
+
+  /**
+   * Renders an A11y compatible tab component.
+   *
+   * Inspired by
+   * - https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-1/tabs.html
+   * - http://web-accessibility.carnegiemuseums.org/code/tabs/
+   */
+  export default defineComponent({
+    name: 'c-tabs',
+
+    // components: {},
+
+    props: {
+      /**
+       * Expects a list of tab definitinos.
+       */
+      tabs: {
+        type: Array as PropType<ITab[]>,
+        required: true,
+      },
+
+      /**
+       * Accepts a label that describes the purpose of the set of tabs.
+       */
+      ariaLabel: {
+        type: String,
```

Comment:
```suggestion
        type: String | null,
```

Reply:
@MB-web this is JS, not TS. But sometimes we use `as PropType`. Not sure what the convention should be. Can we discuss this please?